pom.xml

```diff
@@ -49,8 +49,18 @@
         <id>spark-3.1.1</id>
         <properties>
             <spark.version>${spark311.version}</spark.version>
+            <scala.version>2.12.10</scala.version>
+            <jackson.version>2.10.0</jackson.version>
         </properties>
     </profile>
+    <profile>
+      <id>spark-3.2.0</id>
```

Comment:
question: rename these profiles to `spark-3.2` and `spark-3.1`, as the changes from those minor releases are not so big

Reply:
Yes, make sense. The corresponding minor release may work well also. I will make the change.

Reply:
I tried to compile with Spark 3.2.1, looks like there are indeed some changes. 
`[INFO] /home/sparkuser/git/gazelle_plugin/arrow-data-source/parquet/src/main/scala:-1: info: compiling
[INFO] Compiling 4 source files to /home/sparkuser/git/gazelle_plugin/arrow-data-source/parquet/target/classes at 1646359743584
[ERROR] /home/sparkuser/git/gazelle_plugin/arrow-data-source/parquet/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:344: error: overloaded method constructor VectorizedParquetRecordReader with alternatives:
[INFO]   (x$1: Boolean,x$2: Int)org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader <and>
[INFO]   (x$1: java.time.ZoneId,x$2: String,x$3: String,x$4: String,x$5: String,x$6: Boolean,x$7: Int)org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader
[INFO]  cannot be applied to (java.time.ZoneId, String, String, Boolean, Int)
[INFO]         val vectorizedReader = new VectorizedParquetRecordReader(
[INFO]                                ^
[ERROR] /home/sparkuser/git/gazelle_plugin/arrow-data-source/parquet/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:366: error: type mismatch;
[INFO]  found   : org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy.Value
[INFO]  required: org.apache.spark.sql.catalyst.util.RebaseDateTime.RebaseSpec
[INFO]           convertTz, enableVectorizedReader = false, datetimeRebaseMode, SQLConf.LegacyBehaviorPolicy.LEGACY)
[INFO]                                                      ^
[ERROR] /home/sparkuser/git/gazelle_plugin/arrow-data-source/parquet/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:366: error: type mismatch;
[INFO]  found   : org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy.Value
[INFO]  required: org.apache.spark.sql.catalyst.util.RebaseDateTime.RebaseSpec
[INFO]           convertTz, enableVectorizedReader = false, datetimeRebaseMode, SQLConf.LegacyBehaviorPolicy.LEGACY)
[INFO]                                                                                                       ^
[ERROR] three errors found
`

Reply:
Yes, there are some incompatible issues related to parquet reader after bumping spark versions from 3.2.0 to 3.2.1. I have just fixed all issues found in the compiling. So by now, the project with this patch applied can be successfully built on both spark 3.1.1 & 3.2.1.